### ISSUE TITLE
Top-level avqa package does not re-export documented public API

### ISSUE THREAD

**Issue by sachncs** [sachncs/avqa#18]
## Problem

`README.md` "API Reference" (lines 113-127) lists 13 symbols as public API, and the project docstring in `src/avqa/__init__.py` (lines 11-23) names 11 more. The current `avqa/__init__.py` only exports ~30 of these via `__all__`. Several symbols listed in README are **not** importable from `avqa` directly:

- `AdaptiveRefinement` (class), `MultiPassRefiner`, `RefinementResult` — only in `avqa.refinement`.
- `TopPRouter`, `ThresholdRouter` — only in `avqa.routing`.
- `DefaultScheduler`, `AdaptiveScheduler` — only in `avqa.scheduler`.
- `TorchBackend`, `BackendError` (re-exported), `EuclideanHierarchicalQuantizer`, `VectorQuantizer` (re-exported).
- `Visualizer`, `JSONVisualizer`, `TreeNode`, `HeatmapData`, `TimelineEvent`, `RefinementTreeDict`, `RoutingPathDict`, `CodebookUtilizationDict`, `TimelineDict`.
- `ProfilerReport`, `StageTimer`, `Profiler` (re-exported).
- `InMemoryKVCache`, `PagedKVCache`, `CacheEntry`.
- `hopfield_logits`, `per_query_beta`, `paper_beta`, `validate_adaptive`, `AdaptiveSchedule`, `HopfieldConfig`.
- `online_codebook_adaptation`, `StreamingVQBuffer`, `empty_realisation`.
- `seed_everything`, `DEFAULT_SEED`, `online_softmax_step`, `RankLike`, `ShapeLike`, `coerce_shape`, `shape_to_string`.
- All sub-config dataclasses (`AttentionShapeConfig`, `CodebookConfig`, `RoutingConfig`, `RefinementConfig`, `MergeConfig`, `BackendConfig`, `CacheConfig`, `PrecisionConfig`, `ExecutionConfig`, `HopfieldConfig`, `AVQConfig` (re-exported), plus `require_positive`, `require_non_negative`, `require_in_range`, `to_primitive`, `SCHEMA_VERSION`, `DEFAULT_EMA_DECAY`).
- `SHAPE_*`, `SUPPORTED_DTYPES`, `SUPPORTED_DEVICES`, `TensorContract`, `make_default_contract`, `is_supported_dtype`, `is_supported_device`, `BATCH`, `HEAD`, etc.
- `configure_logger`, `get_logger`, `AVQA_LOGGER_NAME`, `is_configured`, etc.
- `Parse_version`, `__version_info__`, `Codebook` (alias).

## Evidence

`src/avqa/__init__.py:43-114` (current imports) vs. `README.md:113-127` (API table):

```
README claims                                | __init__ exports
---------------------------------------------|--------------------
AdaptiveRefinement (class)                   | ❌ not exported
MultiPassRefiner                             | ✅ exported
TopPRouter                                   | ❌ not exported
ThresholdRouter                              | ❌ not exported
DefaultScheduler                             | ❌ not exported
AdaptiveScheduler                            | ❌ not exported
TorchBackend                                 | ✅ exported
BackendError                                 | ✅ exported
Visualizer                                   | ✅ exported
JSONVisualizer                               | ❌ not exported
TreeNode                                     | ❌ not exported
HeatmapData                                  | ❌ not exported
TimelineEvent                                | ❌ not exported
InMemoryKVCache                              | ❌ not exported
PagedKVCache                                 | ❌ not exported
CacheEntry                                   | ❌ not exported
hopfield_logits                              | ❌ not exported
per_query_beta                               | ❌ not exported
paper_beta                                   | ❌ not exported
online_codebook_adaptation                   | ❌ not exported
StreamingVQBuffer                            | ❌ not exported
seed_everything                              | ❌ not exported
AttentionShapeConfig                         | ❌ not exported
CodebookConfig                               | ❌ not exported
… (etc)                                      |
```

## Root Cause

The README is aspirational about what the project offers; `__init__.py` exports only what `tests/` happen to import. There is no single audit step that ensures the README's API table matches `__all__`.

## Impact

- Users following the README example code get `ImportError`s.
- The package's discoverability via `help(avqa)` is incomplete.
- IDE autocomplete (`avqa.<TAB>`) doesn't surface the additional classes, making the package feel smaller than it is.

## Recommendation

Add a re-export block to `src/avqa/__init__.py` that mirrors the README API table, and add a CI test that diffs `README.md` symbols vs. `avqa.__all__`.

## Acceptance Criteria

- [ ] Every symbol listed in `README.md` lines 113-127 is importable as `from avqa import <Name>`.
- [ ] Add a `tests/test_public_api.py` that asserts the README API table matches `avqa.__all__`.
- [ ] Document in `CONTRIBUTING.md` that adding new public symbols requires both an `__all__` entry and a README API table entry.

## Verification

`python -c "from avqa import AdaptiveRefinement, TopPRouter, DefaultScheduler, seed_everything, hopfield_logits, InMemoryKVCache, TreeNode"` returns without `ImportError`. Run `pytest tests/ -q` to confirm no test imports were broken.

## Risk

Low. Re-exports do not change behavior. The CI test could be skipped with `# doctest: +SKIP` if a symbol legitimately cannot be re-exported.

**Comment by sachncs** [comment]
Resolved in PR #37.
